apis/multiarch/common/plugins/plugins.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugins
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	// PluginName for NodeAffinityScoring.
+	PluginName = "NodeAffinityScoring"
+)
+
+// PluginBase defines shared fields for all plugins
+type PluginBase struct {
+	Enabled bool `json:"Enabled,omitempty" protobuf:"bytes,1,rep,name=Enabled"`
+}
+
+// PlatformConfig holds configuration for specific platforms (like architecture, weight)
+type PlatformConfig struct {
+	Architecture string `json:"architecture,omitempty" protobuf:"bytes,1,rep,name=architecture"`
+	Weight       int    `json:"weight,omitempty" protobuf:"bytes,2,rep,name=weight"`
+}
+
+// NodeAffinityScoringArgs holds the configuration for the scoring plugin
+type NodeAffinityScoringArgs struct {
+	Platforms []PlatformConfig `json:"platforms,omitempty" protobuf:"bytes,2,rep,name=platforms"`
+}
+
+// NodeAffinityScoring is the plugin that implements the ScorePlugin interface
+type NodeAffinityScoring struct {
+	Base PluginBase               `json:"Base,omitempty" protobuf:"bytes,1,opt,name=Base"`
```

Comment:
Let's use embedding to embed the PluginBase fields
```suggestion
	*PluginBase               `json:"Base,omitempty" protobuf:"bytes,1,opt,name=Base"`
```

Reply:
https://gobyexample.com/struct-embedding